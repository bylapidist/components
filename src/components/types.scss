$levels: 'primary', 'secondary', 'tertiary', 'danger';
$sizes: 'none', 'small', 'medium', 'large';
$families: 'body', 'heading', 'serif', 'mono';
$weights: 'regular', 'medium', 'bold';
