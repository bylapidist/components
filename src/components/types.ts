import {
    ElementType,
    ReactNode,
    CSSProperties,
    ComponentPropsWithoutRef
} from 'react';

export type CommonComponentProps<T extends ElementType> = {
    as?: T;
    testId?: string;
    children?: ReactNode;
    className?: string;
    style?: CSSProperties;
};

export type CommonComponentCombinedProps<
    T extends ElementType,
    K
> = CommonComponentProps<T> &
    Omit<ComponentPropsWithoutRef<T>, keyof CommonComponentProps<T>> &
    K;

export type LevelType = 'primary' | 'secondary' | 'tertiary' | 'danger';

export type SizeType = 'none' | 'small' | 'medium' | 'large';
