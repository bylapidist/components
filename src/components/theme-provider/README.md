# @lapidist/theme-provider

![npm](https://img.shields.io/npm/v/@lapidist/theme-provider)
