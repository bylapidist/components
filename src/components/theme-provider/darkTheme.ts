import { Theme } from '@lapidist/styles';

export const darkTheme: Theme = {
    boxShadows: {
        '1':
            '0 1px 3px 0 rgba(255, 255, 255, 0.1), 0 1px 2px 0 rgba(255, 255, 255, 0.06)'
    },
    colors: {
        base: {
            light: 'black',
            dark: 'white',
            transparent: 'transparent'
        },
        grey: {
            lightest: '#292B3E',
            light: '#73748b',
            base: '#d4d4d4',
            dark: '#E8E8E8'
        },
        primary: {
            lightest: '#1a8385',
            light: '#17a3a5',
            base: '#4AD6D8',
            dark: '#9bfff8'
        },
        secondary: {
            lightest: '#c99944',
            light: '#ebc764',
            base: '#fff08f',
            dark: '#fdffb5'
        },
        tertiary: {
            lightest: '#5A8C34',
            light: '#72af5d',
            base: '#A7D981',
            dark: '#e2f7c9'
        },
        danger: {
            lightest: '#810000',
            light: '#b41e25',
            base: '#E75158',
            dark: '#FF848B'
        }
    }
};
