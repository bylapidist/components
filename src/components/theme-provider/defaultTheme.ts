export interface ColorGroup {
    lightest: string;
    light: string;
    base: string;
    dark: string;
}

export interface SizingGroup {
    xxxs: string;
    xxs: string;
    xs: string;
    s: string;
    m: string;
    l: string;
    xl: string;
    xxl: string;
    xxxl: string;
}

export interface BreakpointGroup {
    s: string;
    m: string;
    l: string;
}

export interface DefaultTheme {
    colors: {
        greys: ColorGroup;
        blues: ColorGroup;
        yellows: ColorGroup;
        greens: ColorGroup;
        reds: ColorGroup;
    };
    sizing: SizingGroup;
    breakpoints: {
        mobile: BreakpointGroup;
        tablet: BreakpointGroup;
        laptop: BreakpointGroup;
    };
}

export const greys: ColorGroup = {
    lightest: '#ffffff',
    light: '#f3f5f9',
    base: '#66687d',
    dark: '#292b3e'
};

export const blues: ColorGroup = {
    lightest: '#9bfff8',
    light: '#4AD6D8',
    base: '#17a3a5',
    dark: '#007072'
};

export const yellows: ColorGroup = {
    lightest: '#fdffb5',
    light: '#fff08f',
    base: '#ebc764',
    dark: '#c99944'
};

export const greens: ColorGroup = {
    lightest: '#e2f7c9',
    light: '#A7D981',
    base: '#72af5d',
    dark: '#5A8C34'
};

export const reds: ColorGroup = {
    lightest: '#FF848B',
    light: '#E75158',
    base: '#b41e25',
    dark: '#810000'
};

export const defaultTheme: DefaultTheme = {
    colors: {
        greys,
        blues,
        yellows,
        greens,
        reds
    },
    sizing: {
        xxxs: '2px',
        xxs: '4px',
        xs: '10px',
        s: '14px',
        m: '16px',
        l: '24px',
        xl: '36px',
        xxl: '45px',
        xxxl: '57px'
    },
    breakpoints: {
        mobile: {
            s: '320px',
            m: '375px',
            l: '425px'
        },
        tablet: {
            s: '580px',
            m: '640px',
            l: '768px'
        },
        laptop: {
            s: '1024px',
            m: '1280px',
            l: '1440px'
        }
    }
};

export default defaultTheme;
