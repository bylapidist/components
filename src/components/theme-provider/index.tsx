import React, { Context, PropsWithChildren } from 'react';
import {
    ThemeProvider as StyledThemeProvider,
    createGlobalStyle
} from 'styled-components';
import { mergeThemes, Theme } from '@lapidist/styles';
import { defaultTheme } from './defaultTheme';
import { darkTheme } from './darkTheme';

const GlobalStyle = createGlobalStyle`
    html, body, div, span, applet, object, iframe,
    h1, h2, h3, h4, h5, h6, p, blockquote, pre,
    a, abbr, acronym, address, big, cite, code,
    del, dfn, em, img, ins, kbd, q, s, samp,
    small, strike, strong, sub, sup, tt, var,
    b, u, i, center, button,
    dl, dt, dd, menu, ol, ul, li,
    fieldset, form, label, legend,
    table, caption, tbody, tfoot, thead, tr, th, td,
    article, aside, canvas, details, embed,
    figure, figcaption, footer, header, hgroup,
    main, menu, nav, output, ruby, section, summary,
    time, mark, audio, video {
        margin: 0;
        padding: 0;
        border: 0;
        font-size: 100%;
        font: inherit;
        vertical-align: baseline;
        transition: color 0.12s cubic-bezier(0.6, 0.04, 0.98, 0.7) 0s,
            border 0.12s cubic-bezier(0.6, 0.04, 0.98, 0.7) 0s,
            background-color 0.12s cubic-bezier(0.6, 0.04, 0.98, 0.7) 0s;
    }
    html { box-sizing: border-box; }
    *, *:before, *:after { box-sizing: inherit; }
    article, aside, details, figcaption, figure,
    footer, header, hgroup, main, menu, nav, section { display: block; }
    *[hidden] { display: none; }
    body { line-height: 1; font-size: 62.5%; }
    menu, ol, ul { list-style: none; }
    blockquote, q { quotes: none; }
    blockquote:before, blockquote:after,
    q:before, q:after { content: ''; content: none; }
    table { border-collapse: collapse; border-spacing: 0; }
    img { display: block; max-width: 100%; }
    a, button { text-decoration: none; outline-offset: 2px; outline: transparent solid 2px; }
    a:focus, button:focus { outline: rgb(235, 199, 100) solid 2px; }
`;

export interface ThemeProviderProps extends PropsWithChildren<unknown> {
    readonly theme?: Theme;
}

export const ThemeContext: Context<Theme> =
    React.createContext<Theme>(defaultTheme);

export const useTheme = () => React.useContext(ThemeContext);

export const ThemeProvider: React.FC<ThemeProviderProps> = ({
    children,
    theme
}) => {
    const initialTheme: Theme = useTheme();
    const baseTheme: Theme | undefined = mergeThemes(initialTheme, theme);

    const [isDarkMode, setIsDarkMode] = React.useState<boolean>();
    const [themeContext, setThemeContext] =
        React.useState<Theme | undefined>(baseTheme);

    React.useEffect(() => {
        const isDarkModePersisted =
            localStorage.getItem('isDarkMode') === 'true';
        setIsDarkMode(isDarkModePersisted);
    }, [setIsDarkMode]);

    React.useEffect(() => {
        setThemeContext(
            isDarkMode ? mergeThemes(baseTheme, darkTheme) : baseTheme
        );
    }, [isDarkMode, baseTheme]);

    return (
        <>
            <GlobalStyle />
            <StyledThemeProvider theme={themeContext}>
                {children}
            </StyledThemeProvider>
        </>
    );
};

ThemeProvider.displayName = 'ThemeProvider';
