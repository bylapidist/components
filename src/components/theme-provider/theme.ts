import * as CSS from 'csstype';
import { defaultTheme } from './defaultTheme';

type ThemeObject<T> = { [K: string]: T };

type ThemeArray<T> = { [K: string]: T[] };

export type NumberOrString = string | number | 0;

export type Color = { colorGroup: NumberOrString; colorShade: NumberOrString };

export interface Theme {
    breakpoints?: ThemeObject<string>;
    fontSizes?: ThemeObject<CSS.FontSizeProperty<NumberOrString>>;
    fontFamilies?: ThemeArray<CSS.FontFamilyProperty>;
    fontWeights?: ThemeObject<CSS.FontWeightProperty>;
    lineHeights?: ThemeObject<CSS.LineHeightProperty<NumberOrString>>;
    letterSpacings?: ThemeObject<CSS.LetterSpacingProperty<NumberOrString>>;
    sizes?: ThemeObject<string>;
    borderRadii?: ThemeObject<CSS.BorderRadiusProperty<NumberOrString>>;
    borderWidths?: ThemeObject<CSS.BorderWidthProperty<NumberOrString>>;
    boxShadows?: ThemeObject<CSS.BoxShadowProperty>;
    widths?: ThemeObject<string>;
    colors?: ThemeObject<ThemeObject<CSS.ColorProperty>>;
}

export const mergeThemes = (theme?: Theme): Theme => {
    if (!theme) return defaultTheme;

    return { ...theme, ...defaultTheme };
};

export const getBreakpoint = (
    theme: Theme,
    breakpoint: NumberOrString
): string => {
    if (theme.breakpoints && theme.breakpoints[breakpoint]) {
        return theme.breakpoints[breakpoint];
    }

    throw new Error(`${breakpoint} does not exist on theme.breakpoints`);
};

export const getAllBreakpoints = (theme: Theme): string[] => {
    if (theme.breakpoints) {
        return Object.keys(theme.breakpoints);
    }

    throw new Error(`theme.breakpoints does not exist`);
};

export const getAllMediaQueries = (theme: Theme): ThemeObject<string> => {
    if (theme.breakpoints) {
        const allBreakpoints = getAllBreakpoints(theme);
        const allMediaQueries: ThemeObject<string> = {};

        allBreakpoints.forEach((breakpoint) => {
            allMediaQueries[breakpoint] = `(min-width: ${getBreakpoint(
                theme,
                breakpoint
            )})`;
        });

        return allMediaQueries;
    }

    throw new Error(`theme.breakpoints does not exist`);
};

export const getFontSize = (
    theme: Theme,
    size: NumberOrString
): CSS.FontSizeProperty<NumberOrString> => {
    if (theme.fontSizes && theme.fontSizes[size]) {
        return theme.fontSizes[size];
    }

    throw new Error(`${size} does not exist on theme.fontSizes`);
};

export const getFontFamily = (
    theme: Theme,
    fontFamily: NumberOrString
): CSS.FontFamilyProperty => {
    if (theme.fontFamilies && theme.fontFamilies[fontFamily]) {
        return theme.fontFamilies[fontFamily].join(', ');
    }

    throw new Error(`${fontFamily} does not exist on theme.fontFamilies`);
};

export const getFontWeight = (
    theme: Theme,
    fontWeight: NumberOrString
): CSS.FontWeightProperty => {
    if (theme.fontWeights && theme.fontWeights[fontWeight]) {
        return theme.fontWeights[fontWeight];
    }

    throw new Error(`${fontWeight} does not exist on theme.fontWeights`);
};

export const getLineHeight = (
    theme: Theme,
    lineHeight: NumberOrString
): CSS.LineHeightProperty<NumberOrString> => {
    if (theme.lineHeights && theme.lineHeights[lineHeight]) {
        return theme.lineHeights[lineHeight];
    }

    throw new Error(`${lineHeight} does not exist on theme.lineHeights`);
};

export const getLetterSpacing = (
    theme: Theme,
    letterSpacing: NumberOrString
): CSS.LetterSpacingProperty<NumberOrString> => {
    if (theme.letterSpacings && theme.letterSpacings[letterSpacing]) {
        return theme.letterSpacings[letterSpacing];
    }

    throw new Error(`${letterSpacing} does not exist on theme.letterSpacings`);
};

export const getSize = (theme: Theme, size: NumberOrString): string => {
    if (theme.sizes && theme.sizes[size]) {
        return theme.sizes[size];
    }

    throw new Error(`${size} does not exist on theme.sizes`);
};

export const getBorderRadius = (
    theme: Theme,
    borderRadius: NumberOrString
): CSS.BorderRadiusProperty<NumberOrString> => {
    if (theme.borderRadii && theme.borderRadii[borderRadius]) {
        return theme.borderRadii[borderRadius];
    }

    throw new Error(`${borderRadius} does not exist on theme.borderRadii`);
};

export const getBorderWidth = (
    theme: Theme,
    borderWidth: NumberOrString
): CSS.BorderWidthProperty<NumberOrString> => {
    if (theme.borderWidths && theme.borderWidths[borderWidth]) {
        return theme.borderWidths[borderWidth];
    }

    throw new Error(`${borderWidth} does not exist on theme.borderWidths`);
};

export const getBoxShadow = (
    theme: Theme,
    boxShadow: NumberOrString
): CSS.BoxShadowProperty => {
    if (theme.boxShadows && theme.boxShadows[boxShadow]) {
        return theme.boxShadows[boxShadow];
    }

    throw new Error(`${boxShadow} does not exist on theme.boxShadows`);
};

export const getWidth = (
    theme: Theme,
    width: NumberOrString
): CSS.WidthProperty<NumberOrString> => {
    if (theme.widths && theme.widths[width]) {
        return theme.widths[width];
    }

    throw new Error(`${width} does not exist on theme.widths`);
};

export const getColor = (
    theme: Theme,
    colorGroup: NumberOrString,
    colorShade: NumberOrString
): CSS.ColorProperty => {
    if (
        theme.colors &&
        theme.colors[colorGroup] &&
        theme.colors[colorGroup][colorShade]
    ) {
        return theme.colors[colorGroup][colorShade];
    }

    throw new Error(
        `${colorGroup} or ${colorShade} does not exist on theme.colors`
    );
};
