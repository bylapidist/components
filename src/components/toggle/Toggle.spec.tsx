import * as React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';

import { KindType } from '../types';
import { ThemeProvider } from '../theme-provider';
import { Toggle } from './index';

const setup = (Component: React.ReactElement) =>
    render(<ThemeProvider>{Component}</ThemeProvider>);

afterEach(cleanup);

const kinds: KindType[] = ['primary', 'secondary', 'tertiary', 'danger'];

test.each(kinds)('it works', (kind) => {
    const { container } = setup(<Toggle kind={kind}>Hello world</Toggle>);
    expect(container.firstChild).toMatchSnapshot();
});

test.each(kinds)('it works with checked', (kind) => {
    const { container } = setup(
        <Toggle kind={kind} checked>
            Hello world
        </Toggle>
    );
    expect(container.firstChild).toMatchSnapshot();
});
