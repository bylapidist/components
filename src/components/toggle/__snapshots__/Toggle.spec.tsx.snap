// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  font-size: 1rem;
  font-family: Montserrat,sans-serif;
  line-height: 2;
  font-weight: 400;
  color: #292B3E;
}

.c0 > p {
  margin-bottom: 0.5rem;
}

.c0 > p:last-child {
  margin-bottom: 0;
}

.c1 {
  border-style: solid;
  text-align: center;
  display: inline-block;
  position: relative;
  font-size: 1rem;
  font-weight: 500;
  border-radius: 9999px;
  border-width: 2px;
  box-shadow: 0 2px 3px rgba(0,0,0,0.1);
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  background-color: #17a3a5;
  color: white;
  border-color: #17a3a5;
  border-top-color: transparent;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-left-color: transparent;
}

.c1:hover {
  cursor: pointer;
  background-color: #135f61;
  color: white;
}

.c1:hover:disabled {
  cursor: not-allowed;
  background-color: #135f61;
}

.c1:disabled {
  opacity: 0.7;
  background-color: #135f61;
  color: white;
}

.c2 {
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  class="c0 c1"
>
  <div
    class="c2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-circle "
      data-icon="circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-right: 0.5rem;"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256z"
        fill="currentColor"
      />
    </svg>
    Hello world
  </div>
</button>
`;

exports[`it works 2`] = `
.c0 {
  font-size: 1rem;
  font-family: Montserrat,sans-serif;
  line-height: 2;
  font-weight: 400;
  color: #292B3E;
}

.c0 > p {
  margin-bottom: 0.5rem;
}

.c0 > p:last-child {
  margin-bottom: 0;
}

.c1 {
  border-style: solid;
  text-align: center;
  display: inline-block;
  position: relative;
  font-size: 1rem;
  font-weight: 500;
  border-radius: 9999px;
  border-width: 2px;
  box-shadow: 0 2px 3px rgba(0,0,0,0.1);
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  background-color: white;
  color: #17a3a5;
  border-color: #d4d4d4;
  border-top-color: transparent;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-left-color: transparent;
}

.c1:hover {
  cursor: pointer;
  background-color: #fafafa;
  color: #135f61;
}

.c1:hover:disabled {
  cursor: not-allowed;
  background-color: #fafafa;
}

.c1:disabled {
  opacity: 0.7;
  background-color: #fafafa;
  color: #292B3E;
}

.c2 {
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  class="c0 c1"
>
  <div
    class="c2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-circle "
      data-icon="circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-right: 0.5rem;"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256z"
        fill="currentColor"
      />
    </svg>
    Hello world
  </div>
</button>
`;

exports[`it works 3`] = `
.c0 {
  font-size: 1rem;
  font-family: Montserrat,sans-serif;
  line-height: 2;
  font-weight: 400;
  color: #292B3E;
}

.c0 > p {
  margin-bottom: 0.5rem;
}

.c0 > p:last-child {
  margin-bottom: 0;
}

.c1 {
  border-style: solid;
  text-align: center;
  display: inline-block;
  position: relative;
  font-size: 1rem;
  font-weight: 500;
  border-radius: 9999px;
  border-width: 2px;
  box-shadow: 0 2px 3px rgba(0,0,0,0.1);
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  background-color: white;
  color: #73748b;
  border-color: #fafafa;
  border-top-color: transparent;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-left-color: transparent;
}

.c1:hover {
  cursor: pointer;
  background-color: #fafafa;
  color: #292B3E;
}

.c1:hover:disabled {
  cursor: not-allowed;
  background-color: #fafafa;
}

.c1:disabled {
  opacity: 0.7;
  background-color: #fafafa;
  color: #292B3E;
}

.c2 {
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  class="c0 c1"
>
  <div
    class="c2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-circle "
      data-icon="circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-right: 0.5rem;"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256z"
        fill="currentColor"
      />
    </svg>
    Hello world
  </div>
</button>
`;

exports[`it works 4`] = `
.c0 {
  font-size: 1rem;
  font-family: Montserrat,sans-serif;
  line-height: 2;
  font-weight: 400;
  color: #292B3E;
}

.c0 > p {
  margin-bottom: 0.5rem;
}

.c0 > p:last-child {
  margin-bottom: 0;
}

.c1 {
  border-style: solid;
  text-align: center;
  display: inline-block;
  position: relative;
  font-size: 1rem;
  font-weight: 500;
  border-radius: 9999px;
  border-width: 2px;
  box-shadow: 0 2px 3px rgba(0,0,0,0.1);
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  background-color: white;
  color: #b41e25;
  border-color: #810000;
  border-top-color: transparent;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-left-color: transparent;
}

.c1:hover {
  cursor: pointer;
  background-color: #fafafa;
  color: #810000;
}

.c1:hover:disabled {
  cursor: not-allowed;
  background-color: #fafafa;
}

.c1:disabled {
  opacity: 0.7;
  background-color: #fafafa;
  color: #292B3E;
}

.c2 {
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  class="c0 c1"
>
  <div
    class="c2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-circle "
      data-icon="circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-right: 0.5rem;"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256z"
        fill="currentColor"
      />
    </svg>
    Hello world
  </div>
</button>
`;

exports[`it works with checked 1`] = `
.c0 {
  font-size: 1rem;
  font-family: Montserrat,sans-serif;
  line-height: 2;
  font-weight: 400;
  color: #292B3E;
}

.c0 > p {
  margin-bottom: 0.5rem;
}

.c0 > p:last-child {
  margin-bottom: 0;
}

.c1 {
  border-style: solid;
  text-align: center;
  display: inline-block;
  position: relative;
  font-size: 1rem;
  font-weight: 500;
  border-radius: 9999px;
  border-width: 2px;
  box-shadow: 0 2px 3px rgba(0,0,0,0.1);
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  background-color: #17a3a5;
  color: white;
  border-color: #17a3a5;
  border-top-color: #73748b;
  border-right-color: #73748b;
  border-bottom-color: #73748b;
  border-left-color: #73748b;
}

.c1:hover {
  cursor: pointer;
  background-color: #135f61;
  color: white;
}

.c1:hover:disabled {
  cursor: not-allowed;
  background-color: #135f61;
}

.c1:disabled {
  opacity: 0.7;
  background-color: #135f61;
  color: white;
}

.c2 {
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  class="c0 c1"
>
  <div
    class="c2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-circle-dot "
      data-icon="circle-dot"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-right: 0.5rem;"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM256 352C309 352 352 309 352 256C352 202.1 309 160 256 160C202.1 160 160 202.1 160 256C160 309 202.1 352 256 352z"
        fill="currentColor"
      />
    </svg>
    Hello world
  </div>
</button>
`;

exports[`it works with checked 2`] = `
.c0 {
  font-size: 1rem;
  font-family: Montserrat,sans-serif;
  line-height: 2;
  font-weight: 400;
  color: #292B3E;
}

.c0 > p {
  margin-bottom: 0.5rem;
}

.c0 > p:last-child {
  margin-bottom: 0;
}

.c1 {
  border-style: solid;
  text-align: center;
  display: inline-block;
  position: relative;
  font-size: 1rem;
  font-weight: 500;
  border-radius: 9999px;
  border-width: 2px;
  box-shadow: 0 2px 3px rgba(0,0,0,0.1);
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  background-color: white;
  color: #17a3a5;
  border-color: #d4d4d4;
  border-top-color: #73748b;
  border-right-color: #73748b;
  border-bottom-color: #73748b;
  border-left-color: #73748b;
}

.c1:hover {
  cursor: pointer;
  background-color: #fafafa;
  color: #135f61;
}

.c1:hover:disabled {
  cursor: not-allowed;
  background-color: #fafafa;
}

.c1:disabled {
  opacity: 0.7;
  background-color: #fafafa;
  color: #292B3E;
}

.c2 {
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  class="c0 c1"
>
  <div
    class="c2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-circle-dot "
      data-icon="circle-dot"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-right: 0.5rem;"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM256 352C309 352 352 309 352 256C352 202.1 309 160 256 160C202.1 160 160 202.1 160 256C160 309 202.1 352 256 352z"
        fill="currentColor"
      />
    </svg>
    Hello world
  </div>
</button>
`;

exports[`it works with checked 3`] = `
.c0 {
  font-size: 1rem;
  font-family: Montserrat,sans-serif;
  line-height: 2;
  font-weight: 400;
  color: #292B3E;
}

.c0 > p {
  margin-bottom: 0.5rem;
}

.c0 > p:last-child {
  margin-bottom: 0;
}

.c1 {
  border-style: solid;
  text-align: center;
  display: inline-block;
  position: relative;
  font-size: 1rem;
  font-weight: 500;
  border-radius: 9999px;
  border-width: 2px;
  box-shadow: 0 2px 3px rgba(0,0,0,0.1);
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  background-color: white;
  color: #73748b;
  border-color: #fafafa;
  border-top-color: #73748b;
  border-right-color: #73748b;
  border-bottom-color: #73748b;
  border-left-color: #73748b;
}

.c1:hover {
  cursor: pointer;
  background-color: #fafafa;
  color: #292B3E;
}

.c1:hover:disabled {
  cursor: not-allowed;
  background-color: #fafafa;
}

.c1:disabled {
  opacity: 0.7;
  background-color: #fafafa;
  color: #292B3E;
}

.c2 {
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  class="c0 c1"
>
  <div
    class="c2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-circle-dot "
      data-icon="circle-dot"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-right: 0.5rem;"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM256 352C309 352 352 309 352 256C352 202.1 309 160 256 160C202.1 160 160 202.1 160 256C160 309 202.1 352 256 352z"
        fill="currentColor"
      />
    </svg>
    Hello world
  </div>
</button>
`;

exports[`it works with checked 4`] = `
.c0 {
  font-size: 1rem;
  font-family: Montserrat,sans-serif;
  line-height: 2;
  font-weight: 400;
  color: #292B3E;
}

.c0 > p {
  margin-bottom: 0.5rem;
}

.c0 > p:last-child {
  margin-bottom: 0;
}

.c1 {
  border-style: solid;
  text-align: center;
  display: inline-block;
  position: relative;
  font-size: 1rem;
  font-weight: 500;
  border-radius: 9999px;
  border-width: 2px;
  box-shadow: 0 2px 3px rgba(0,0,0,0.1);
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  background-color: white;
  color: #b41e25;
  border-color: #810000;
  border-top-color: #73748b;
  border-right-color: #73748b;
  border-bottom-color: #73748b;
  border-left-color: #73748b;
}

.c1:hover {
  cursor: pointer;
  background-color: #fafafa;
  color: #810000;
}

.c1:hover:disabled {
  cursor: not-allowed;
  background-color: #fafafa;
}

.c1:disabled {
  opacity: 0.7;
  background-color: #fafafa;
  color: #292B3E;
}

.c2 {
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  class="c0 c1"
>
  <div
    class="c2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-circle-dot "
      data-icon="circle-dot"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="margin-right: 0.5rem;"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM256 352C309 352 352 309 352 256C352 202.1 309 160 256 160C202.1 160 160 202.1 160 256C160 309 202.1 352 256 352z"
        fill="currentColor"
      />
    </svg>
    Hello world
  </div>
</button>
`;
